src/ray_serializer.jl

```diff
@@ -110,7 +139,9 @@ function deserialize_from_ray_object(ray_obj::SharedPtr{ray_jll.RayObject},
     end
 
     for inner_object_ref in s.object_refs
-        _register_ownership(inner_object_ref, outer_object_ref)
+        metadata = s.object_owner[inner_object_ref]
+        _register_ownership(inner_object_ref, outer_object_ref, metadata.owner_address,
+                            metadata.serialized_object_status)
```

Comment:
could even do property name destructuring here if you wanted (I dunno if it makes things any clearer or not though)

Reply:
I'll take a quick look to see if that makes things more readable

Reply:
Seems about the same. I'll update this to use the destructuring for no real reason beyond I want to

Reply:
😎